Crashlytics/Crashlytics/Controllers/FIRCLSRolloutsPersistenceManager.m

```diff
@@ -58,10 +58,15 @@ - (void)updateRolloutsStateToPersistenceWithRollouts:(NSData *_Nonnull)rollouts
   NSFileHandle *rolloutsFile = [NSFileHandle fileHandleForUpdatingAtPath:rolloutsPath];
 
   dispatch_sync(FIRCLSGetLoggingQueue(), ^{
-    [rolloutsFile seekToEndOfFile];
-    [rolloutsFile writeData:rollouts];
-    NSData *newLineData = [@"\n" dataUsingEncoding:NSUTF8StringEncoding];
-    [rolloutsFile writeData:newLineData];
+    @try {
+      [rolloutsFile seekToEndOfFile];
+      NSMutableData *rolloutsWithNewLineData = [rollouts mutableCopy];
+      [rolloutsWithNewLineData appendData:[@"\n" dataUsingEncoding:NSUTF8StringEncoding]];
+      [rolloutsFile writeData:rolloutsWithNewLineData];
+    } @catch (NSException *exception) {
+      FIRCLSDebugLog(@"Failed to write new rollouts. Exception name: %s - message: %s",
```

Comment:
Could we make this an error log so it shows up in red?

Reply:
`FIRCLSDebugLog` is leveraged on Core `FIRLogBasic` so might be good to keep everything in consistent (as they are following [apple system logging](https://opensource.apple.com/source/Libc/Libc-391.5.22/include/asl.h.auto.html)).